0000-open-options.md

```diff
@@ -0,0 +1,636 @@
+not end up in the read cache, so this can have a performance penalty when
+reading if a file is opened with a read-write access mode.
+
+
+#### Sync data
+`.sync_data(true)`
+
+Some systems support syncing only the data written, but can wait with updating
+less critical metadata such as the last modified timestamp. If the metadata is
+not critical (and it rarely is), you should always use `sync_data()` as an easy
+performance win.
+
+Linux since version 2.6.33 supports this mode with `O_DSYNC`, as does Solaris
+and recent versions of NetBSD. If a system does not support only syncing data,
+Rust will fall back to full syncing.
+
+If `.sync_all(true)` is specified, `.sync_data()` is ignored.
+
+
+### Completely bypass the kernel
+Normally the os kernel will process read or write calls and store the data
+temporarily in a kernel-space buffer. The kernel makes sure the data size and
+alignment of reads and writes correspondent to sectors on the storage device,
+usually 512 or 4096 bytes. Also the kernel can keep data recently read or
+written in cache, to speed up future file operations.
+
+Some operating systems allow you to completely bypass the copy of data to or
+from kernel space. This is generally a bad idea. Applications will have to
+figure out and handle alignment restrictions themselves, and implement manual
+caching. It is mostly useful for database applications that may have more
+knowledge about their optimal caching behaviour than the os. And it can have a
+use when reading many gigabytes of data (like a backup process), which may
+destroy the os cache for other processes.
+
+This is available on Windows with the flag `FILE_FLAG_NO_BUFFERING`, and on
+Linux and some variants of BSD with `O_DIRECT`. Making correct use of this mode
+involves low-level tuning and operating system dependant behaviour. It makes no
+sense for Rust to expose this as a simple, cross-platform option. For
+applications that really wish to use it, it is no problem to submit it as a
+custom flag.
+
+
+## Asynchronous IO
+Out op scope.
+
+
+## Other options
+
+### Inheritance of file descriptors
+Leaking file descriptors to child processes can cause problems and can be a
+security vulnerability. See this report by 
+[Python](https://www.python.org/dev/peps/pep-0446/).
+
+On Windows, child processes do not inherit file descriptors by default (but this
+can be changed). On Unix they always inherit, unless the close-on-exec flag is
+set.
+
+The close on exec flag can be set atomically when opening the file, or later
+with `fcntl`. The `O_CLOEXEC` flag is in the relatively new POSIX-2008 standard,
+and all modern versions of Unix support it. The following table lists for which
+operating systems we can rely on the flag to be supported.
+
+os            | since version | oldest supported version
+:-------------|:--------------|:------------------------
+OS X          | 10.6          | 10.7?
+Linux         | 2.6.23        | 2.6.32 (supported by Rust)
```

Comment:
At the time I write this part, `O_CLOEXEC` was not yet passed, but now it is not really an interesting part of the rfc anymore :)